assessments/unit-01/code.js

```diff
@@ -5,3 +5,48 @@ const emit = (a, b, product) => {
 };
 
 // Write your code here ...
+const averageWeight = (t, n) => t / n
+const hypotenuse = (a, b) => Math.sqrt(a ** 2 + b ** 2)
+const maxRadius = (w, h) => Math.min(w, h) / 2
+const numCircles = (r, w) => Math.floor(w / (2 * r))
+const offset = (a, f) => (a - f) / 2
+const canSleepIn = (weekday, vacation) => !weekday || vacation
+const canGoToProm = (senior, invited, excluded) => (senior && !excluded) || (invited && !excluded)
+const getsSpeedingTicket = (s, grouchy) => (s > 65 && grouchy) || (s > 70 && !grouchy)
+const moreThanTwiceAsLong = (x, y) => (x / y) > 2
+const aFartherThanB = (a, b, c) => Math.abs(a - c) > Math.abs(b - c)
+const firstHalf = (s) => s.substring(0, Math.floor(s.length / 2))
+const secondHalf = (s) => s.substring(Math.floor(s.length / 2))
+const upDown = (s) => s.toUppperCase() + s.toLowerCase()
+const everyOther = (s) => s.substring(0, 1) + s.substring(2, 3) + s.substring(4, 5)
+const upDownLastCharacter = (s) => s.substring(length - 1).toUpperCase + s.substring(length - 1).toLowerCase
```

Comment:
Where you have `length` it needs to be `s.length`. And you need `()`s after `toUpperCase` and `toLowerCase` the same as you had in `upDown`.

This is also another case where you could also use `s[s.length - 1]` rather than using `substring`.

For extra style (nobody did it on this assessment, I don't think) you could use the `upDown` function you already wrote:

 ```javascript
const upDownLastCharacter = (s) => upDown(s[s.length - 1])
```